docs/debugging.asciidoc

```diff
@@ -0,0 +1,23 @@
+[[debugging]]
+== Debugging the agent itself
+
+Hopefully the agent Just Works™ for you but depending on your situation the agent could need some tuning.
+
+Firstly, to know more about what's going on inside the agent, you can increase the amount of log messages it writes by setting the log level with the option `log_level = 0` -- `0` being the level of most messages, `DEBUG`.
+
+[float]
+[[debugging-errors]]
+=== Errors
+
+[float]
+[[debugging-errors-queue-full]]
+==== `Queue is full (256 items), skipping…`
+
+The agent has an internal queue that holds events after they are done, before they are safely serialized and sent to APM Server. To avoid using up all of your memory, the queue has a fixed size. Taking from the queue happens faster than adding to it but it can happen that the queue fills, hence the warning.
+
+Things to consider:
+
+  - Is `server_url` misconfigured or APM Server down?
```

Comment:
If the URL is misconfigured or the server is down, will the user see other errors too? Perhaps list them out and refer to them here, so the user can manually correlate?